### PR TITLE
Refactor cmd/nerdctl, part 2

### PR DESCRIPTION
**Pull request by apostasie**
This is on top of #3379 and continuation of the work started there.

As the previous PR, this does not contain any logic change or fixes - this is purely refactoring.

This one deals with:
- container
- compose
- leftovers (inspect, system)

### CODE REVIEW

**Comment by apostasie** [comment]
@AkihiroSuda @Zheaoli 

CI is green. This is ready - obviously though, #3379 and #3394 need to go first.

**Comment by apostasie** [comment]
Will rebase once #3379 gets in.

**Comment by apostasie** [comment]
Rebased.
Pending green CI of course.

File: cmd/nerdctl/main.go [hunk 1]
```diff
@@ -440,36 +421,3 @@ func AddPersistentBoolFlag(cmd *cobra.Command, name string, aliases, nonPersiste
 		}
 	}
 }
-
-// AddPersistentStringArrayFlag is similar to cmd.Flags().StringArray but supports aliases and env var and persistent.
```
Comment (AkihiroSuda) [inline cmd/nerdctl/main.go:444]: AddPersistentStringFlag should be moved too

Comment (apostasie) [reply cmd/nerdctl/main.go:444]: We should, of course.
It is only used by `main.go` though (and so is `AddPersistentBoolFlag`).
There is also a fair amount of cleanup to do in `helpers`.

What about we take care of these (and further scrubbing the new helpers directory) in a follow-up PR?

Comment (apostasie) [reply cmd/nerdctl/main.go:444]: Changed my mind with more coffee beating the lazyness out of me.

Added a new commit that take care of post-cleanup:
- moved the last Flag methods
- moved some linux specific stuff from `testing.go` to `testing_linux.go`
- removed now empty `main_unix.go`

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**